conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/ReturnTypeMapper.java

```diff
@@ -44,4 +46,16 @@ public TypeName baseType(Optional<Type> type) {
     public TypeName async(Optional<Type> type) {
         return ParameterizedTypeName.get(LISTENABLE_FUTURE, Primitives.box(baseType(type)));
     }
+
+    public boolean isBinaryOrOptionalBinary(TypeName returnType) {
```

Comment:
do these need to be public?

Reply:
They don't. Neither do the other methods in this class. Can make a separate PR for that nit.

Reply:
Opened #2472